### ISSUE TITLE
Showtooltip showing picture, but not tooltip

### ISSUE THREAD

**Issue by Ketrel** [brittyazel/Neuron#339]
For a couple of my macros, for some reason, the tooltip comes up as 'Unknown'

This is one  
```
#showtooltip
/use [btn:1] Reflecting Prism
/cancelaura [btn:2] Prismatic Reflection
```

This is another  
```  
#showtooltip Hook of the Master Angler
/equip [btn:1] Hook of the Master Angler
/use [btn:1] Hook of the Master Angler
/cancelaura [btn:2] Master Angler
```

I tried with the full `[button:1]` notation as well with no difference.  
It does pick up the icon, just not the tooltip itself.

**Comment by brittyazel** [comment]
I think this may be fixed now. Can you check the latest code from MASTER and report back?

**Comment by Ketrel** [comment]
Just a straight swap to the current Master didn't change anything (still says Unknown and shows no tooltip other than the word)
If I re-add the macro the the bar, it picks up the name of the macro instead of the word 'Unknown' and it's in yellow.

**Comment by Ketrel** [comment]
Hmm, a couple more things. (With the latest Master)

* The yellow name of the macro is now showing up for ANY I macro I add, even ones that worked before.  
* None of the macros with modifiers (newly added or on bars prior) change their icon or tooltip now and only show the macro name for newly added, or the default nomod action for ones from before.

**Comment by brittyazel** [comment]
Do you happen to have any druid monk or shaman macros that you can replicate this with and paste here? It would help me diagnose this. I rewrote the macro parser entirely, so it's quite possible there are edge cases that I for sure missed

**Comment by Ketrel** [comment]
Two of the examples are toys and the third is a trinket, and all three utilize the `/use` command vs a `/cast` I'm not sure how I'd replicate that as class's spells.

**Comment by brittyazel** [comment]
Hrm I don't have any of those toys.

I tried with this one:
#showtooltip
/use [btn:1] Crashin' Thrashin' Juggernaught
/cancelaura [btn:2] Crashin' Thrashin' Juggernaught

And it worked perfectly for me :-/ I see the full icon and tooltip

**Comment by brittyazel** [comment]
Question, are you using Neuron to write the macros are are these macros in the default Blizzard UI?

**Comment by Ketrel** [comment]
I do have the Crashin' Thrashin' Juggernaught, so I tried with that macro.
I get the same 'Unknown' as with the others.

As far as that part, I'm creating them in the Blizzard UI and dragging them to the bar from Neuron.

**Comment by Ketrel** [comment]
Quick followup, I tried making it directly on the Neuron button, but that's even worse.  If I don't chose an icon, it's invisible, if I choose any other, I get the "?" icon, and it still says 'Unknown' for the tooltip.

I should mention that in this and my previous comment's case, I'm back on the current release version.

**Comment by brittyazel** [comment]
Yeah it doesn't surprise me that the current release version does that. It's super broken with non-standard macros, hence why I gutted it and rewrote it.

So where it stands right now, at least for me on the MASTER branch is that this macro will work great as a Neuron Macro, and as a Blizzard macro it's only showing the name in Yello, bot an actual tooltip.

**Comment by brittyazel** [comment]
Ok blizzard macros should now show the name in yellow stably. They don't show the true tooltips though because we are not parsing the contents of the Blizzard macros, so we don't know what the spell is this we can't form a tooltip

**Comment by Ketrel** [comment]
Two related but separate questions.

For how it is now, would the macro still function however?  Would a modifier key still work for example or is that lost as well?

The other, is if this isn't solvable some other way, why not make it duplicate the macro if a Blizzard macro is dragged to a slot?  It would read the Blizzard one only that once and save the contents of the macro as its own.  That would seem to be a good compromise to still give full functionality with drag/drop.

**Comment by brittyazel** [comment]
Modifier keys should still work. The macro would be used exactly the same as it always would be.

As for the second option, that's totally doable, the issue then is that if someone updated the Blizz macro it wouldn't then update for Neuron and they may get confused. Though, in theory, I could have the Neuron button reparse each and every time the Blizz macro changed... You could then end up with the reverse issue though, which is if they update the macro it Neuron it wouldn't update the Blizz macro lol.

**Comment by Ketrel** [comment]
That second case though is fine as far as I'm concerned, because to get a fully functional macro, that's what already happens, since you have to make it through Neuron.  I was thinking even reparsing on update wouldn't be doable.  I don't know about to everyone, but that seems like the perfect compromise.  Copy the blizz one on add, reparse on update.  

And if I'm understanding right, the only way to go out of sync is to edit it directly, but then again, at that point, it really IS a separate macro, so this seems like ideal behavior.

**Comment by brittyazel** [comment]
Cool, I can definitely make this happen :-)
